src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -122,15 +119,14 @@ private void drawJoints(final Graphics2D graphics, final Point[] points, final i
    * @param scale The scale-factor of the Map
    */
   private void drawCustomCursor(final Graphics2D graphics, final RouteDescription routeDescription, final int xOffset,
-      final int yOffset,
-      final double scale, final Dimension imageDimension, final Dimension screenDimension) {
+      final int yOffset, final double scale) {
     final Image cursorImage = routeDescription.getCursorImage();
     if (cursorImage != null) {
-      final Point wrappedEndPoint =
-          getPointOnMap(routeDescription.getEnd(), xOffset, yOffset, imageDimension, screenDimension);
-      graphics.drawImage(cursorImage,
-          (int) (((wrappedEndPoint.x - xOffset) - (cursorImage.getWidth(null) / 2)) * scale),
-          (int) (((wrappedEndPoint.y - yOffset) - (cursorImage.getHeight(null) / 2)) * scale), null);
+      for (Point[] endPoint : routeOptimizer.getAllPoints(routeOptimizer.getLastEndPoint())) {
+        graphics.drawImage(cursorImage,
+            (int) (((endPoint[0].x - xOffset) - (cursorImage.getWidth(null) / 2)) * scale),
+            (int) (((endPoint[0].y - yOffset) - (cursorImage.getHeight(null) / 2)) * scale), null);
```

Comment:
What does this do? 
`cursorImage.getHeight(null)`

Can we avoid passing in null to a method and having it do some seemingly magic work? This could be a useful thing to move to a library/utility somewhere so when reading it we are not left wondering how that works (or what the `null` param does)